internal/graph/dispatch_throttling_check_resolver_test.go

```diff
@@ -0,0 +1,105 @@
+package graph
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/goleak"
+)
+
+type countResolveCheck struct {
+	count *atomic.Uint32
+}
+
+func newCountResolveCheck() *countResolveCheck {
+	return &countResolveCheck{
+		count: &atomic.Uint32{},
+	}
+}
+
+func (c *countResolveCheck) ResolveCheck(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error) {
+	c.count.Add(1)
+	return &ResolveCheckResponse{Allowed: true}, nil
+}
+
+func (c *countResolveCheck) Close() {}
+
+func (c *countResolveCheck) NumResolveCheck() uint32 {
+	return c.count.Load()
+}
+
+func TestDispatchThrottlingCheckResolver(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	t.Run("below_shaping_level_should_not_be_shaped", func(t *testing.T) {
```

Comment:
This test name and the one below don't make sense to me - it's not obvious what shaping is referring to here.

I may suggest `dispatch_below_threshold_resolves_immediately`.

Reply:
updated